src/manage.c

```diff
@@ -5605,22 +5605,74 @@ credential_full_type (const char* abbreviation)
 /* System reports. */
 
 /**
- * @brief Get system report types from a slave.
+ * @brief Get a performance report from an OSP scanner.
+ *
+ * @param[in]  scanner          The scanner to get the performance report from.
+ * @param[in]  start            The start time of the performance report.
+ * @param[in]  end              The end time of the performance report.
+ * @param[in]  titles           The end titles for the performance report.
+ * @param[in]  performance_str  The performance string.
+ *
+ * @return 0 if successful, 4 could not connect to scanner,
+ *         6 failed to get performance report, -1 error
+ */
+static int
+get_osp_performance_string (scanner_t scanner, int start, int end,
+                            const char *titles, gchar **performance_str)
+{
+  char *host, *ca_pub, *key_pub, *key_priv;
+  int port;
+  osp_connection_t *connection;
+  osp_get_performance_opts_t opts;
+  gchar *error;
+
+  host = scanner_host (scanner);
+  port = scanner_port (scanner);
+  ca_pub = scanner_ca_pub (scanner);
+  key_pub = scanner_key_pub (scanner);
+  key_priv = scanner_key_priv (scanner);
+
+  connection = osp_connect_with_data (host, port, ca_pub, key_pub, key_priv);
+
+  if (connection == NULL)
+    return 4;
+
+  opts.start = start;
+  opts.end = end;
+  opts.titles = g_strdup (titles);
+  error = NULL;
+
+  if (osp_get_performance_ext (connection, opts, performance_str, &error))
+    {
+      osp_connection_close (connection);
+      g_warning ("Error getting OSP performance report: %s", error);
+      g_free (error);
+      g_free (opts.titles);
+      return 4;
+    }
+
+  osp_connection_close (connection);
+  g_free (opts.titles);
+
+  return 0;
+}
+
+/**
+ * @brief Get system report types from a GMP slave.
  *
  * @param[in]   required_type  Single type to limit types to.
  * @param[out]  types          Types on success.
  * @param[out]  start          Actual start of types, which caller must free.
- * @param[out]  slave_id       ID of GMP slave.
+ * @param[out]  slave          GMP slave.
  *
- * @return 0 if successful, 2 failed to find slave, 3 unused, 4 could not
+ * @return 0 if successful, 2 unused, 3 unused, 4 could not
```

Comment:
This "unused" business is confusing.  Can we either remove the return, or
describe it better, like "No slave given"?

Reply:
I've removed the unused return values from the docs and made the functions return -1 if `slave` is 0 (because the check should be done by the caller) in fa06038573cd2d5a8dd6668db77ea32b84aba92d.